registry/application/services/validate_service_assets.py

```diff
@@ -0,0 +1,184 @@
+import json
+import os
+
+from common import utils
+from common.boto_utils import BotoUtils
+from common.logger import get_logger
+from registry.config import REGION_NAME, DEMO_COMPONENT_CODE_BUILD_NAME, \
+    UPDATE_DEMO_COMPONENT_BUILD_STATUS_LAMBDA_ARN, MANAGE_PROTO_COMPILATION_LAMBDA_ARN, ALLOWED_HERO_IMAGE_FORMATS
+from registry.infrastructure.repositories.service_publisher_repository import ServicePublisherRepository
+
+logger = get_logger(__name__)
+boto_utils = BotoUtils(region_name=REGION_NAME)
+service_repo = ServicePublisherRepository()
+
+
+class ValidateServiceAssets:
+    def __init__(self):
+        pass
+
+    @staticmethod
+    def validate_service_assets(payload):
+        uploaded_file_path = payload["Records"][0]['s3']['object']['key']
+        bucket_name = payload["Records"][0]['s3']['bucket']['name']
+        name, extension = os.path.splitext(uploaded_file_path)
+        if uploaded_file_path.endswith('_component.zip'):
+            return ValidateServiceAssets.trigger_demo_component_code_build(uploaded_file_path, bucket_name)
+        elif uploaded_file_path.endswith('_proto_files.zip'):
+            ValidateServiceAssets.validate_proto(uploaded_file_path, bucket_name)
+        elif extension in ALLOWED_HERO_IMAGE_FORMATS:
+            ValidateServiceAssets.update_hero_image_url(uploaded_file_path, bucket_name)
+
+    @staticmethod
+    def trigger_demo_component_code_build(file_path, bucket_name):
+        try:
```

Comment:
The scope of try-except is too big. It should be limited to the execution of trigger code build statement.